### PR TITLE
Api 37483 v1 2122 apt unit number

### PR DESCRIPTION
**Pull request by mchristiansonVA**
## Summary

- *This work is behind a feature toggle (flipper): NO*
- Updates v1 2122 schema aptUnitNumber attribute to accept one of integer or string, to allow our customers to submit field values containing numeric or alphanumeric data.

## Related issue(s)
[API-37483](https://jira.devops.va.gov/browse/API-37483)

## Testing done

- [x] *New code is covered by unit tests*
- Confirmed via local testing in Postman that alphanumeric values supplied to the aptUnitNumber attribute are accepted as valid.
- Added rspec test for same.

Example request:

```
{
    "data": {
        "type": "form/21-22",
        "attributes": {
            "veteran": {
                "address": {
                    "numberAndStreet": "76 Crowther Ave",
                    "aptUnitNumber": "11C",
                    "city": "Bridgeport",
                    "country": "US",
                    "state": "CT",
                    "zipFirstFive": "06616"
                }
            },
            "serviceOrganization": {
                "poaCode": "067"
            }
        }
    }
}
```

## What areas of the site does it impact?
modules/claims_api/config/schemas/v1/2122.json
modules/claims_api/spec/requests/v1/power_of_attorney_request_spec.rb

## Acceptance criteria

- [x]  I fixed|updated|added unit tests and integration tests for each feature (if applicable).
- [x]  No error nor warning in the console.
- [ ]  Events are being sent to the appropriate logging solution
- [ ]  Documentation has been updated (link to documentation)
- [x]  No sensitive information (i.e. PII/credentials/internal URLs/etc.) is captured in logging, hardcoded, or specs
- [ ]  Feature/bug has a monitor built into Datadog or Grafana (if applicable)
- [ ]  If app impacted requires authentication, did you login to a local build and verify all authenticated routes work as expected
- [ ]  I added a screenshot of the developed feature

### CODE REVIEW

File: modules/claims_api/config/schemas/v1/2122.json [hunk 1]
```diff
@@ -30,7 +30,12 @@
...
             },
-            "aptUnitNumber":   { "type": "integer" },
+            "aptUnitNumber": {
+              "oneOf": [
+                { "type": "integer" },
+                { "type":"string" }
+              ]
+            },
             "city": {
               "description": "City for the address.",
```
Comment (stiehlrod) [inline modules/claims_api/config/schemas/v1/2122.json:40]: @mchristiansonVA this seems weird to me, like the city and aptUnitNumber are one attribute

**Review by rockwellwindsor-va — APPROVED** [review]
Those updated docs looked good to me.  Was abe to submit with both string and integer in the aptUnitNumber field

Screenshot 2024-06-17 at 2 32 31 PM

File: modules/claims_api/config/schemas/v1/2122.json (hunk 1 above)
Comment (mchristiansonVA) [reply modules/claims_api/config/schemas/v1/2122.json:40]: I neglected to add the generated swagger, so latest updates should show this better. Note that the screenshot shows the original implementation, which doesn't have space between the aptUnitNumber and city attributes.

**Review by rockwellwindsor-va — APPROVED** [review]
Docs looked good with update.  Submissions with combinations of integers and strings all worked as expected.

Screenshot 2024-06-18 at 11 07 21 AM

**Pull request merged**